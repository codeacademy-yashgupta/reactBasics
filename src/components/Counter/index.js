import React, { Component } from 'react';
import Button from './../Button/'
class Counter extends Component {
    state = {
      count: 10
    }
    increaseCounter = () => {
      console.log("Hello");
      this.setState({count: this.state.count + 5})
    }
    decreaseCounter = () => {
      console.log("hi")
      this.setState({count: this.state.count - 5})
    }
  render() {
    
    return (
      
      <div className="divButton">
        Counter: {this.state.count}
        <Button caption="Add 5" onClick={this.increaseCounter}></Button>
        <Button caption="Decrease 5" onClick={this.decreaseCounter}></Button>
      </div>
    );
  }
}

export default Counter;
