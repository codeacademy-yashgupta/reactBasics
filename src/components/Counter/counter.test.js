import React from 'react';
import Counter from './';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';


describe('Counter', () => {
  it('should render a button', () => {
    const tree = renderer.create(<Counter/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should initialize the value with zero.', () => {
    const wrapper = shallow(<Counter></Counter>);
    expect(wrapper.instance().state.count).toEqual(10);
  })
  it('should increment counter on increaseCount function', () => {
    const wrapper = shallow(<Counter></Counter>);
    wrapper.instance().increaseCounter();
    expect(wrapper.instance().state.count).toEqual(15);
  });
  it('should decrease counter on decsreaseCount function', () => {
    const wrapper =shallow (<Counter></Counter>);
    wrapper.instance().decreaseCounter();
    expect(wrapper.instance().state.count).toEqual(5);
  });
  it('should set the value with initial value when both functions are called once (decrease and increase count)', () => {
    const wrapper = shallow (<Counter></Counter>);
    wrapper.instance().decreaseCounter();
    wrapper.instance().increaseCounter();
    expect(wrapper.instance().state.count).toEqual(10);
  });
});