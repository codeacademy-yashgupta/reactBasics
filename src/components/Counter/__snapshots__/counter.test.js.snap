// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render a button 1`] = `
<div
  className="divButton"
>
  Counter: 
  10
  <div
    className="divButton"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Add 5
    </button>
  </div>
  <div
    className="divButton"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Decrease 5
    </button>
  </div>
</div>
`;
